### ISSUE TITLE
`TSVD()` as compression algorithm produces method error

### ISSUE THREAD

**Issue by RmMetsch** [IntegralEquations/HMatrices.jl#73]
Hello, 

I'm trying to use HMatrices to compress a Matrix using TSVD. I followed the tutorial from the docs page and swapped out the compression algorithm `PartialACA()` for `TSVD()` but get the following error : 

MethodError: no method matching (::TSVD)(::HMatrices.PermutedMatrix{…}, ::ClusterTree{…}, ::ClusterTree{…}, ::HMatrices.ACABuffer{…})

not sure what to do, any help is appreciated.

```
using HMatrices, StaticArrays

const Point2D = SVector{2,Float64}
# points on a circle
m = n = 10_000
X = Y = [Point2D(sin(i*2π/n),cos(i*2π/n)) for i in 0:n-1]

struct LaplaceMatrix <: AbstractMatrix{Float64}
    X::Vector{Point2D}
    Y::Vector{Point2D}
end

Base.getindex(K::LaplaceMatrix,i::Int,j::Int) = -1/2π*log(norm(K.X[i] - K.Y[j]) + 1e-10)
Base.size(K::LaplaceMatrix) = length(K.X), length(K.Y)

# create the abstract matrix
K = LaplaceMatrix(X,Y)

Xclt = Yclt = ClusterTree(X)
adm = StrongAdmissibilityStd()
comp = TSVD()

H = assemble_hmatrix(K,Xclt,Yclt;adm,comp,threads=false,distributed=false)
```

**Comment by RmMetsch** [comment]
Hey @maltezfaria, no worries we are all busy nowadays :)

I ran ] update HMatrices and  the code above and it still produces the same error :/ 

I'm using it to compress a mutual inductance matrix of a electrical cable. The geometry is rather fixed so I only have to compute it once. A colleague of mine insisted I use this method for better accuracy of the compression. 

Cheers,
Roel

**Comment by maltezfaria** [comment]
You need to be on the `main` branch since I have not (yet) tagged a new version with the fix. `] update` will only update to a stable version.

**Comment by maltezfaria** [comment]
New tagged version is up. Running `] update` should work now. Let me know if that is not the case. 

Regarding your application: why not use the adaptive cross approximation (ACA) compression method in `HMatrices` (the default)? It is usually much faster than e.g. SVD, at the cost of being less robust...

**Comment by RmMetsch** [comment]
Thanks for updating, I did some testing and while ACA compression is indeed much faster the TSVD compression has superior accuracy. As I said, I only have to compute it once so its worth the wait.

**Comment by maltezfaria** [comment]
Fair enough, but keep in mind you can pass a tolerance parameter to all of these methods, so you can probably get good results from `ACA` by passing a more strict tolerance. But if `TSVD` suffices for you, great 👍 . I just never optimized it since I use that mostly for validation purposes.

**Comment by maltezfaria** [comment]
One last thing in defense of `TSVD` I guess: it is the *optimal low rank* approximation (in Frobenius or spectral norm) and the only *guaranteed* method in `HMatrices`.
